db/migrate/20230119073650_create_post.rb

```diff
@@ -0,0 +1,13 @@
+class CreatePost < ActiveRecord::Migration[7.0]
+  def change
+    create_table :posts do |t|
+      t.references :users, null: false, foreign_key: true
```

Comment:
- [ ]  according to project requirement kindly reference user with author i.e `t.references :author, foreign_key: { to_table: 'users' } `